File: src/backend/Tasa_de_Retencion.js
import React, { useState } from 'react';

function CalcularTasaRetencion() {
    const [clientesInicio, setClientesInicio] = useState(5000);
    const [nuevosClientes, setNuevosClientes] = useState(1200);
    const [clientesFinal, setClientesFinal] = useState(4200);
    const [tasaRetencion, setTasaRetencion] = useState(0);

    const Tasa_de_Retencion = () => {
        if (clientesInicio === 0) {
            setTasaRetencion(0); // Evita división por cero
        } else {
            const tasaRetencionCalc = ((clientesFinal - nuevosClientes) / clientesInicio) * 100;
            setTasaRetencion(tasaRetencionCalc.toFixed(2));
        }
    };

    return (
        <div>
            <h2>Calculadora de Tasa de Retención de Clientes</h2>
            <form>
                <label>
                    Clientes al inicio del período:
                    <input
                        type="number"
                        value={clientesInicio}
                        onChange={(e) => setClientesInicio(parseInt(e.target.value, 10))}
                    />
                </label>
                <br />
                <label>
                    Nuevos clientes durante el período:
                    <input
                        type="number"
                        value={nuevosClientes}
                        onChange={(e) => setNuevosClientes(parseInt(e.target.value, 10))}
                    />
                </label>
                <br />
                <label>
                    Clientes al final del período:
                    <input
                        type="number"
                        value={clientesFinal}
                        onChange={(e) => setClientesFinal(parseInt(e.target.value, 10))}
                    />
                </label>
                <br />
                <button onClick={handleCalculate}>Calcular Tasa de Retención</button>
            </form>
            <p>
                La tasa de retención de clientes es: {tasaRetencion}%
            </p>
        </div>
    );
}

export default CalcularTasaRetencion;

File: src/backend/Tasa_de_Retencion.py
def calcular_tasa_retencion(clientes_inicio, nuevos_clientes, clientes_final):
    """
    Calcula la tasa de retención de clientes.

    Args:
        clientes_inicio (int): Número de clientes al inicio del período.
        nuevos_clientes (int): Número de nuevos clientes durante el período.
        clientes_final (int): Número de clientes al final del período.

    Returns:
        float: Tasa de retención de clientes en porcentaje.
    """
    if clientes_inicio == 0:
        return 0  # Evita división por cero

    tasa_retencion = ((clientes_final - nuevos_clientes) / clientes_inicio)*100
    return tasa_retencion

# Ejemplo de uso
clientes_inicio = 5000
nuevos_clientes = 1200
clientes_final = 5200

tasa_retencion = calcular_tasa_retencion(clientes_inicio, nuevos_clientes, clientes_final)
print(f"La tasa de retención de clientes es: {tasa_retencion:.2f}")